pkg/controllers/secretstore/common.go

```diff
@@ -84,14 +85,22 @@ func validateStore(ctx context.Context, namespace string, store esapi.GenericSto
 		return fmt.Errorf(errStoreProvider, err)
 	}
 
+	if !storeProvider.SupportsConcurrency() {
+		mu.Lock()
+	}
 	cl, err := storeProvider.NewClient(ctx, store, client, namespace)
```

Comment:
same here: will deadlock if `NewClient` fails.

Reply:
I believe this is why the e2e tests are failing actually 😓 I would take ages to realize that. thanks!